lib/loader/loader.c

```diff
@@ -87,30 +87,28 @@ evmc_create_fn evmc_load(const char* filename, enum evmc_loader_error_code* erro
     if (strncmp(name_pos, lib_prefix, lib_prefix_length) == 0)
         name_pos += lib_prefix_length;
 
-    strcpy_s(name + prefix_length, PATH_MAX_LENGTH, name_pos);
+    char* base_name = prefixed_name + prefix_length;
+    strcpy_s(base_name, PATH_MAX_LENGTH, name_pos);
 
     // Trim the file extension.
-    char* ext_pos = strrchr(name, '.');
+    char* ext_pos = strrchr(prefixed_name, '.');
     if (ext_pos)
         *ext_pos = 0;
 
     // Replace all "-" with "_".
-    char* dash_pos = name;
+    char* dash_pos = base_name;
     while ((dash_pos = strchr(dash_pos, '-')) != NULL)
         *dash_pos++ = '_';
 
-    // Search for the "full name" based function name.
-    create_fn = DLL_GET_CREATE_FN(handle, name);
-    if (!create_fn)
+    // Search for the built function name.
+    while ((create_fn = DLL_GET_CREATE_FN(handle, prefixed_name)) == NULL)
     {
-        // Try the "short name" based function name.
-        const char* short_name_pos = strrchr(name, '_');
-        if (short_name_pos)
-        {
-            short_name_pos += 1;
-            memmove(name + prefix_length, short_name_pos, strlen(short_name_pos) + 1);
-            create_fn = DLL_GET_CREATE_FN(handle, name);
-        }
+        // Shorten the base name by skipping the `word_` segment.
+        const char* shorter_name_pos = strchr(base_name, '_');
+        if (!shorter_name_pos)
+            break;
+
+        memmove(base_name, shorter_name_pos + 1, strlen(shorter_name_pos) + 1);
     }
 
     if (!create_fn)
```

Comment:
True. I will check it out.

Reply:
No, it will get messy. I would need to add another loop terminating condition (like strlen == 0) and we don't want to check for "evmc_create_" but "evmc_create".

Also, the naming convention will change soon and we might want to try "evmc_create" first.